src/main/java/com/liberty/social/handler/RatelimitExceptionHandlingController.java

```diff
@@ -6,12 +6,12 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 
 @ControllerAdvice
-public class ExceptionHandlingControllerAdvice {
+public class RatelimitExceptionHandlingController {
```

Comment:
Rename to ExceptionHandlingController, because it will handle all exceptions.
Try to handle 404 exception (Create NotFaundException(String) and handle it here)